src/Sentry/Laravel/Tracing/Integrations/LighthouseIntegration.php

```diff
@@ -0,0 +1,174 @@
+            return;
+        }
+
+        $this->operations[] = [$startExecution->operationName, $operationDefinition];
+
+        $context = new SpanContext;
+        $context->setOp(
+            sprintf(
+                'graphql.%s{%s}',
+                $operationDefinition->operation,
+                $startExecution->operationName ?? implode(',', $this->extractOperationNames($operationDefinition))
+            )
+        );
+
+        $this->operationSpan = $this->requestSpan->startChild($context);
+
+        SentrySdk::getCurrentHub()->setSpan($this->operationSpan);
+    }
+
+    public function handleEndExecution(EndExecution $endExecution): void
+    {
+        if ($this->operationSpan === null) {
+            return;
+        }
+
+        $this->operationSpan->finish();
+        $this->operationSpan = null;
+
+        SentrySdk::getCurrentHub()->setSpan($this->requestSpan);
+    }
+
+    public function handleEndRequest(EndRequest $endRequest): void
+    {
+        if ($this->requestSpan === null) {
+            return;
+        }
+
+        $this->requestSpan->finish();
+        $this->requestSpan = null;
+
+        SentrySdk::getCurrentHub()->setSpan($this->previousSpan);
+        $this->previousSpan = null;
+
+        $this->updateTransaction();
+
+        $this->operations = [];
+    }
+
+    private function updateTransaction(): void
+    {
+        $groupedOperations = [];
+
+        foreach ($this->operations as [$operationName, $operation]) {
+            if (!isset($groupedOperations[$operation->operation])) {
+                $groupedOperations[$operation->operation] = [];
+            }
+
+            if ($operationName === null) {
+                $groupedOperations[$operation->operation] = array_merge(
+                    $groupedOperations[$operation->operation],
+                    $this->extractOperationNames($operation)
+                );
+            } else {
+                $groupedOperations[$operation->operation][] = $operationName;
+            }
+        }
+
+        array_walk($groupedOperations, static function (array &$operations, string $operationType) {
+            sort($operations, SORT_STRING);
+
+            $operations = "{$operationType}{" . implode(',', $operations) . '}';
+        });
+
+        ksort($groupedOperations, SORT_STRING);
+
+        $transactionName = 'lighthouse?' . implode('&', $groupedOperations);
+
+        Integration::setTransaction($transactionName);
+        SentrySdk::getCurrentHub()->getTransaction()->setName($transactionName);
+    }
+
+    private function extractOperationNames(OperationDefinitionNode $operation): array
+    {
+        if ($operation->name !== null) {
+            return [$operation->name->value];
+        }
+
+        $selectionSet = [];
+
+        /** @var \GraphQL\Language\AST\FieldNode $selection */
+        foreach ($operation->selectionSet->selections as $selection) {
+            $selectionSet[] = $selection->alias === null
```

Comment:
```suggestion
            // Not respecting aliases because they are only relevant for clients
            // and the tracing we extract here is targeted at server developers.
            $selectionSet[] = $selection->name->value;
```